gen-cmd/main.go

```diff
@@ -0,0 +1,208 @@
+// generator for the boilerplate code for the endpoint
+// to generate the code anotate the endpoint definition with
+//		//go:generate go run ../gen-cmd/main.go packageName
+//		//go:generate go fmt
+// and run `go generate`
+package main
+
+import (
+	"fmt"
+	"go/ast"
+	"go/parser"
+	"go/token"
+	"os"
+	"strings"
+	"text/template"
+	"unicode"
+)
+
+const (
+	contextType = "context.Context"
+)
+
+func main() {
+	pck := os.Args[1]
+	f := buildAst(pck)
+
+	vs := &Visitor{
+		Package: pck,
+
+		// function helpers
+		ReturnExp: ReturnExp,
+		ArgsExp:   ArgsExp,
+		CompoExp:  CompoExp,
+	}
+
+	ast.Walk(vs, f)
+
+	// TODO remove relative path
+	mergeTemplate(vs, "definition", "../gen-cmd/endpoint_definition.tmpl")
+}
+
+func buildAst(pck string) *ast.File {
+	definitions := pck + ".go"
```

Comment:
Not nice to get the file from the packageName requires the convention of the file `packageName.go`